esp/services/ldapenvironment/ldapenvironmentService.cpp

```diff
@@ -0,0 +1,204 @@
+                throw MakeStringException(ECLWATCH_INVALID_SEC_MANAGER, MSG_SEC_MANAGER_IS_NULL);
+        }
+        
+        //Verify request
+
+        if (isEmptyString(req.getEnvName()))
+        {
+            throw MakeStringException(-1, "Environment name must be specified");
+            if (strchr(req.getEnvName(), ' '))
+                throw MakeStringException(-1, "Environment name cannot contain spaces or special characters");
+        }
+        if (isEmptyString(req.getEnvOwnerName()))
+            throw MakeStringException(-1, "Environment Owner name must be specified");
+        if (isEmptyString(req.getEnvDescription()))
+            throw MakeStringException(-1, "Environment Description must be specified");
+
+        if (req.getFilesMode()==2  && isEmptyString(req.getCustomFilesBaseDN()))
+            throw MakeStringException(-1, "CustomFilesBaseDN must be specified (ex. 'ou=files,ou=hpcc,dc=myLDAPDomain,dc=com')");
+        if (!req.getGroupsMode()==2 && isEmptyString(req.getCustomGroupsBaseDN()))
+            throw MakeStringException(-1, "CustomGroupsBaseDN must be specified (ex. 'ou=groups,ou=hpcc,dc=myLDAPDomain,dc=com')");
+        if (!req.getUsersMode()==2 && isEmptyString(req.getCustomUsersBaseDN()))
+            throw MakeStringException(-1, "CustomUsersBaseDN must be specified (ex. 'ou=users,ou=hpcc,dc=myLDAPDomain,dc=com')");
+        if (!req.getResourcesMode()==2 && isEmptyString(req.getCustomResourcesBaseDN()))
+            throw MakeStringException(-1, "CustomResourcesBaseDN must be specified (ex. 'ou=smc,ou=espservices,ou=hpcc,dc=myLDAPDomain,dc=com')");
+        if (!req.getWorkunitsMode()==2 && isEmptyString(req.getCustomWorkunitsBaseDN()))
+            throw MakeStringException(-1, "CustomWorkunitsBaseDN must be specified (ex. 'ou=workunits,ou=hpcc,dc=myLDAPDomain,dc=com')");
+
+        // Create OU names
+
+        StringBuffer respFilesBaseDN, respGroupsBaseDN, respUsersBaseDN, respResourcesBaseDN, respWorkunitsBaseDN;
+        formatOUname(respFilesBaseDN, req.getEnvName(), req.getFilesMode(), sharedFilesBaseDN.str(), req.getCustomFilesBaseDN(), "ou=files");
+        formatOUname(respGroupsBaseDN, req.getEnvName(), req.getGroupsMode(), sharedGroupsBaseDN.str(), req.getCustomGroupsBaseDN(), "ou=groups");
+        formatOUname(respUsersBaseDN, req.getEnvName(), req.getUsersMode(), sharedUsersBaseDN.str(), req.getCustomUsersBaseDN(), "ou=users");
+        formatOUname(respResourcesBaseDN, req.getEnvName(), req.getResourcesMode(), sharedResourcesBaseDN.str(), req.getCustomResourcesBaseDN(), "ou=smc,ou=espservices");
+        formatOUname(respWorkunitsBaseDN, req.getEnvName(), req.getWorkunitsMode(), sharedWorkunitsBaseDN.str(), req.getCustomWorkunitsBaseDN(), "ou=workunits");
+
+        // Create HPCC Admin Username
+
+        VStringBuffer respAdminUser("admin_%s", req.getEnvName());
+
+
+        //Create LDAP resources OU Hierarchy
+
+        if (req.getCreateLDAPEnvironment())
```

Comment:
Is setting this to false just a way to get the ldapHelm even after it's been created?